### DISCUSSION TITLE
How to access Ocelot config outside of Ocelot pipeline?

### DISCUSSION THREAD

**Discussion by rwestonbutler** [ThreeMammals/Ocelot#1871] [Q&A] [answered]
Is there a way to access the strongly typed Ocelot configuration from outside of the Ocelot pipeline in a delegating handler or middleware, maybe have it injected via DI?

I've run into a situation where I need to grab a config value based on a part of the upstream path.

**Comment by margaale** [comment]
We have the issue #738 exactly for that, is gonna take a while though...

**Comment by rwestonbutler** [comment]
Gotcha, thank you!

**Comment by raman-m** [comment]
Hey @rwestonbutler and @margaale!
How were your holidays? 😉 
Pay attention that #738 is accepted but it is still in development waiting for development.
But it describes a bit different scenario with custom properties and seems **custom properties will not be strongly typed**.

**Comment by raman-m** [comment, marked as answer]
@rwestonbutler [on Jan 16, 2019](https://github.com/ThreeMammals/Ocelot/discussions/1871#discussion-6021428):
> Is there a way to access the strongly typed Ocelot configuration from outside of the Ocelot pipeline **in a delegating handler or middleware, maybe have it injected via DI?**

Please try to inject the following interface objects:
- `IInternalConfiguration`: see [usage and code](https://github.com/search?q=repo%3AThreeMammals%2FOcelot%20IInternalConfiguration&type=code). It has `Routes` property and you have to find your route mannually.
- `IOptionsMonitor<IInternalConfiguration>`: see [usage and code](https://github.com/search?q=repo%3AThreeMammals%2FOcelot%20IOptionsMonitor%3CIInternalConfiguration%3E&type=code). But it requires to write valid path to extract.
- `IOptions<FileConfiguration>`: see [usage and code](https://github.com/search?q=repo%3AThreeMammals%2FOcelot%20IOptions%3CFileConfiguration%3E&type=code). Seems it is the best option!

Hope it helps!
Let us know your testing results please!
